stats.go

```diff
@@ -34,3 +34,109 @@ func Median(input []float64) (output float64) {
 
 	return output
 }
+
+// Max returns the maximum value of the sample
+func Max(input []float64) float64 {
+	if len(input) == 0 {
+		return math.NaN()
+	}
+	max := input[0]
+	for i := 1; i < len(input); i++ {
+		if input[i] > max {
+			max = input[i]
+		}
+	}
+	return max
+}
+
+// Min returns the minimum value of the sample
+func Min(input []float64) float64 {
+	if len(input) == 0 {
+		return math.NaN()
+	}
+	min := input[0]
+	for i := 1; i < len(input); i++ {
+		if input[i] < min {
+			min = input[i]
+		}
+	}
+	return min
+}
+
+// Range returns the difference between the largest and smallest values
+func Range(input []float64) float64 {
+	if len(input) < 2 {
+		return math.NaN()
+	}
+	return Max(input) - Min(input)
+}
+
+// sumOfSquaredDifferences returns the sum of the squared differences of each observation from the mean
+func sumOfSquaredDifferences(input []float64) float64 {
+	if len(input) < 2 {
+		return math.NaN()
+	}
+	mean := Mean(input)
+	ssd := 0.0
+	for _, o := range input {
+		ssd += math.Pow(o-mean, 2.0)
+	}
+	return ssd
+}
+
+// StandardDeviationSample returns the standard deviation of the sample
+func StandardDeviationSample(input []float64) float64 {
+	return math.Sqrt(VarianceSample(input))
+}
+
+// VarianceSample returns the variance of the sample
+func VarianceSample(input []float64) float64 {
+	num := sumOfSquaredDifferences(input)
+	den := float64(len(input) - 1)
+	return num / den
```

Comment:
Is it faster if you call the two parts inline, like `return sumOfSquaredDifferences(input) / float64(len(input) - 1)`?

Reply:
Yes! Performance is improved.
From:
```
BenchmarkVarianceSample0-32             1000000000               3.20 ns/op            0 B/op          0 allocs/op
BenchmarkVarianceSample1-32             1000000000               2.91 ns/op            0 B/op          0 allocs/op
BenchmarkVarianceSample2-32             69475694                86.8 ns/op             0 B/op          0 allocs/op
BenchmarkVarianceSample10-32            13512231               405 ns/op               0 B/op          0 allocs/op
BenchmarkVarianceSample1e3-32             120768             49563 ns/op               0 B/op          0 allocs/op
BenchmarkVarianceSample1e6-32                118          52834614 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/orvend/stats 100.480s
```
To:
```
BenchmarkVarianceSample0-32             1000000000               2.81 ns/op            0 B/op          0 allocs/op
BenchmarkVarianceSample1-32             1000000000               2.75 ns/op            0 B/op          0 allocs/op
BenchmarkVarianceSample2-32             76150934                77.6 ns/op             0 B/op          0 allocs/op
BenchmarkVarianceSample10-32            15028654               376 ns/op               0 B/op          0 allocs/op
BenchmarkVarianceSample1e3-32             120529             46184 ns/op               0 B/op          0 allocs/op
BenchmarkVarianceSample1e6-32                128          49593414 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/orvend/stats 98.807s
```